grpc-protocol/src/main/java/com/linecorp/armeria/common/grpc/protocol/ArmeriaMessageFramer.java

```diff
@@ -188,10 +206,7 @@ private ByteBuf write(ByteBuf message, boolean compressed, boolean webTrailers)
         final int messageLength = message.readableBytes();
         if (maxOutboundMessageSize >= 0 && messageLength > maxOutboundMessageSize) {
             message.release();
-            throw new ArmeriaStatusException(
-                    StatusCodes.RESOURCE_EXHAUSTED,
-                    String.format("message too large %d > %d", messageLength,
-                                  maxOutboundMessageSize));
+            throw newMessageTooLargeException(messageLength);
```

Comment:
Curious if we need to check against the decoded data or the encoded data.

Reply:
We were checking the size of compressed data when compressing which means the actual size. So I followed it by checking the size of encoded data. 😉